src/presign/participant.rs

```diff
@@ -44,36 +44,60 @@ use crate::{
 use libpaillier::unknown_order::BigNumber;
 use merlin::Transcript;
 use rand::{CryptoRng, RngCore};
-use serde::{Deserialize, Serialize};
 use std::collections::HashMap;
 use tracing::{error, info, instrument};
 
-/// Storage identifiers for the presign protocol.
-#[derive(Clone, Copy, Debug, Serialize)]
-#[serde(tag = "Presign")]
-pub(crate) enum StorageType {
-    Ready,
-    RoundOnePrivate,
-    RoundOnePublic,
-    RoundOnePublicBroadcast,
-    RoundTwoPrivate,
-    RoundTwoPublic,
-    RoundThreePrivate,
-    RoundThreePublic,
-    Record,
-}
+// Local storage data types.
+mod storage {
+    use crate::local_storage::TypeTag;
 
-impl Storable for StorageType {}
+    pub(super) struct Ready;
+    impl TypeTag for Ready {
+        type Value = ();
+    }
+    pub(super) struct RoundOnePrivate;
+    impl TypeTag for RoundOnePrivate {
+        type Value = crate::presign::round_one::Private;
+    }
+    pub(super) struct RoundOnePublic;
+    impl TypeTag for RoundOnePublic {
+        type Value = crate::presign::round_one::Public;
+    }
+    pub(super) struct RoundOnePublicBroadcast;
+    impl TypeTag for RoundOnePublicBroadcast {
+        type Value = crate::presign::round_one::PublicBroadcast;
+    }
+    pub(super) struct RoundTwoPrivate;
+    impl TypeTag for RoundTwoPrivate {
+        type Value = crate::presign::round_two::Private;
+    }
+    pub(super) struct RoundTwoPublic;
+    impl TypeTag for RoundTwoPublic {
+        type Value = crate::presign::round_two::Public;
+    }
+    pub(super) struct RoundThreePrivate;
+    impl TypeTag for RoundThreePrivate {
+        type Value = crate::presign::round_three::Private;
+    }
+    pub(super) struct RoundThreePublic;
+    impl TypeTag for RoundThreePublic {
+        type Value = crate::presign::round_three::Public;
+    }
+}
 
-#[derive(Serialize, Deserialize, Clone, Debug)]
+#[derive(Debug)]
 pub(crate) struct PresignParticipant {
     /// A unique identifier for this participant
     id: ParticipantIdentifier,
     /// A list of all other participant identifiers participating in the
     /// protocol
     other_participant_ids: Vec<ParticipantIdentifier>,
+    /// Old storage mechanism currently used to store persistent data
+    ///
+    /// TODO #205: To be removed once we remove the need for persistent storage
```

Comment:
nit: This should reference #180, not #205. 180 removes main storage from `ProtocolParticipant`s. 205 removes it from `Participant`.

Reply:
Yes good catch! I updated the other references to 205 as well.